src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerUploadFileServiceImpl.java

```diff
@@ -1,42 +1,225 @@
+                    combinedErrorMessage.append(KFSConstants.NEWLINE);
+                }
+                combinedErrorMessage.append(buildErrorMessageStringForReport(errorMessage));
+            }
+            results.setErrorMessage(combinedErrorMessage.toString());
+        }
+    }
+    
+    private SupplierResponseMessage buildSupplierResponseMessage(String responseString) {
+        SupplierSyncMessage synchMessage;
+        try {
+            synchMessage = cuMarshalService.unmarshalStringIgnoreDtd(responseString, SupplierSyncMessage.class);
+        } catch (JAXBException | XMLStreamException | IOException e) {
+            LOG.error("buildSupplierResponseMessage, got an error creating SupplierSyncMessage from the response string", e);
+            throw new RuntimeException(e);
+        }
+        SupplierResponseMessage responseMessage = (SupplierResponseMessage) synchMessage.getSupplierSyncMessageItems().get(0);
+        return responseMessage;
+    }
+    
+    private String buildErrorMessageStringForReport(ErrorMessage errorMessage) {
+        String type = StringUtils.trim(errorMessage.getType());
+        String message = StringUtils.trim(errorMessage.getValue());
+        return MessageFormat.format(configurationService.getPropertyValueAsString(CUPurapKeyConstants.JAGGAER_UPLOAD_XML_ERROR_MESSAGE), type, message);
+    }
+    
+    private void processUnsuccessfulResponse(JaggaerUploadFileResultsDTO results, int numberOfAttempts,
+            String responseString, int responseStatusCode) {
+        LOG.error("processUnsuccessfulResponse, attempt number {}, had an unsuccessful webservice call.  Response status was {}", 
+                numberOfAttempts, responseString);
+        results.setFileProcessedByJaggaer(false);
+        results.setResponseCode(String.valueOf(responseStatusCode));
+        results.setErrorMessage(configurationService.getPropertyValueAsString(CUPurapKeyConstants.JAGGAER_UPLOAD_WEBSERVICE_ERROR));
+    }
 
-        for (String fileName : fileNamesToUpload) {
-            LOG.info("uploadSupplierXMLFiles. processing {}", fileName);
-            byte[] uploadFileData = LoadFileUtils.safelyLoadFileBytes(fileName);
-            uploadFile(uploadFileData);
+    private URI buildSupplierUploadURI() {
+        String url = getParameterValueString(CUPurapParameterConstants.JAGGAER_UPLOAD_ENDPOINT);
+        try {
+            return new URI(url);
+        } catch (URISyntaxException e) {
+            LOG.error("buildSupplierUploadURI(): URL: " + url, e);
+            throw new RuntimeException(e);
+        }
+    }
+    
+    private void waitBetweenTries() {
+        try {
+            TimeUnit.SECONDS.sleep(5);
+        } catch (InterruptedException e) {
+            LOG.error("waitBetweenTries, had an error waiting", e);
+        }
+    }
+    
+    private void generateResultsReport(List<JaggaerUploadFileResultsDTO> resultsList) {
+        /*
+         * this will be implemented as a report on Jira 
```

Comment:
Do you need this comment? If so, I would suggest updating it to mention the specific JIRA ticket that will be implementing the report.